### ISSUE TITLE
Issue: Sporadically issue with Zowe editor (An unknown error occured: undefined)

### ISSUE THREAD

**Issue by lv198** [zowe/zlux#981]
**Describe the bug**
Sporadically after stopping/starting Zowe, I get the following error message in the Zowe editor:
"An unknown error occured: undefined" as seen in the screenshot . The issue can only be resolved with restarting Zowe (sometimes several times). In the Zowe log, I get the following message: failed to verify signature with status 53817348 - Signature is not correct

**Steps to Reproduce**
Still not really sure as it only occurs sporadically.

**Expected behavior**
Editor should work as expected.

**Screenshots (if needed)**
![grafik](https://github.com/zowe/zlux/assets/89929118/11e44b86-3d13-4f79-87b8-48cf1f0447a6)

**Logs**

I can get Logs if needed...

**Describe your environment**

- Zowe version number (Check the Desktop login screen, or manifest.json in the Zowe install folder): 2.11.0
- Install method (pax, smpe, kubernetes, github clone): SMPE
- Operating system (z/OS, kubernetes, etc) and OS version: z/OS 3.1
- Node.js version number (Shown in logs, or via `node --version`): v18.18.2
- Java version number (Shown in logs, or via `java -version`): 1.8.0_341
- z/OSMF version: 3.1
- Browser: Mozilla Firefox
- Are you accessing the Desktop from the APIML Gateway? (Recommended): YES
- What is the output of log message ZWES1014I: ZWES1014I ZIS status - 'Ok' (name='ZWESIS_STD      ', cmsRC='0', description='Ok', clientVersion='2')
- Environment variables in use:

**Additional context**

**Comment by 1000TurquoisePogs** [comment]
Please open the developer console of the browser (usually, f12) and check the network tab.
Whenever this issue occurs, see if there is a failed network request associated with the failure.
This looks like a failure to reach ZSS, for some reason, but the screenshot does not say why.

**Comment by lv198** [comment]
Here is a censored screenshot after the issue occured:
![grafik](https://github.com/zowe/zlux/assets/89929118/128f8e07-ca63-4f9f-b3ec-b46be839609e)

**Comment by lv198** [comment]
@1000TurquoisePogs is the screenshot helpful? Do you have any idea how to solve the problem?

**Comment by 1000TurquoisePogs** [comment]
Hi yes it is helpful.
You are getting HTTP 401 on calls to the ZSS server.
That happens when ZSS cannot verify your login session - such as an expired cookie or no cookie in the browser, or maybe some new issue.
Try clicking the user icon at the bottom right of the desktop to do a logout, then log back in. Does it change the behavior?

**Comment by lv198** [comment]
Hi @1000TurquoisePogs 
performing a logout and logging in again does not change the behaviour - even deleting all Browser Data (cookies etc.) is not helpful. The only possibility to solve this is restating Zowe and even that does not solve the problem with a probability of 100% - sometimes it takes more than one restart until it works again.

What else can I check?

**Comment by Sam-Explorer** [comment]
Hi, Did your issue with Zowe editor got fixed?
Was it working initially, any thing need to be checked in zowe.yaml for Zowe Editor to work ?

**Comment by 1000TurquoisePogs** [comment]
Hi apologies for the long wait on this one but, are either of you using a high-availability configuration in which there might be more than one copy of ZSS or app-server running at the same time? We had bugs in the past where cookies would not be usable between multiple servers, causing one to be okay with a request, and the other to think of it as unauthenticated. That particular bug is to be fixed in v2.15.0 releasing shortly.

**Comment by Sam-Explorer** [comment]
Hi, thank you for replying, we are not using high-availability configuration.
however when I check the logs I see the below messages

BPXTLS failed: rc=-1, return code=139, reason code=0x0be802af
for this I found update in - https://github.com/zowe/docs-site/issues/705  - Was thinking to try 

also, I see ZWED0070I - User=undefined (org.zowe.zlux.auth.trivial): Session authenticate successful

MVS-Explorer and USS-Explorer works, its just MVS-Editor which has issue ' An unknown error occurred: undefined '.

In Developer console of the browser, I see
 Http failure response for https://zowetest.com: 403 OK
Refused to apply style from 'https://zowetest.com' because its MIME type ('application/json') is not a supported stylesheet MIME type, and strict MIME checking is enabled.


Any advice on this issue?

Thanks

**Comment by 1000TurquoisePogs** [comment]
Okay that's a very interesting error you get.
reason code according to IBM documentation is:

02AF	JREnvDirty

The specified function is not supported in an address space where a load was done that is not program controlled.

Action: Make sure that programs being loaded into this address space are defined as program controlled. 


And, ZSS, the server that's being used, is a program-controlled server. So, why did it lose program control status? Not sure but, I'm going to open an issue there to research further.
https://github.com/zowe/zss/issues/692

**Comment by 1000TurquoisePogs** [comment]
By chance, is the file `zowe/components/zss/bin/zssServer` NOT program controlled ( https://www.ibm.com/docs/en/zos/2.1.0?topic=authorizations-program-control )
ZSS must be, and as the doc recommends the way to add it would be `extattr +p zowe/components/zss/bin/zssServer` but, it is distributed in that form so I normally would expect it already has that set.

**Comment by Sam-Explorer** [comment]
Exactly, we did refer to the below link and checked it seems to be ok
https://www.ibm.com/docs/en/zos/2.2.0?topic=installation-establishing-program-control

**Comment by 1000TurquoisePogs** [comment]
Not sure if this is whats happening to you, but I learned that some programs can interfere with the program control state of another running program such as Zowe.
Do you see any ICH or BPX messages in the logs / syslog about program control?
For example,
ICH422I THE ENVIRONMENT CANNOT BECOME UNCONTROLLED.               
BPXP014I ENVIRONMENT MUST REMAIN CONTROLLED FOR SERVER (BPX.SERVER)
PROCESSING.

**Comment by Sam-Explorer** [comment]
Hi, thank you very much for helping on the issue and sorry for the delay in reply.
After we updated the file owner of Zowe to Zowe ID and recycle the Zowe STC issue got fixed. 
Now we are able to access the Zowe Editor.

**Comment by JoeNemo** [comment]
So is this resolved as a config/privilege issue?

@lv198 - is this resolution satisfactory??

**Comment by JoeNemo** [comment]
@1000TurquoisePogs can you elaborate on the program trying to "uncontrol" zowe?

**Comment by lv198** [comment]
> So is this resolved as a config/privilege issue?
> 
> @lv198 - is this resolution satisfactory?? 

Hello @1000TurquoisePogs 

Yes solution is satisfactory, was a privilege problem.
Thank you for your help.

**Comment by rudatp** [comment]
> Hi, thank you very much for helping on the issue and sorry for the delay in reply. After we updated the file owner of Zowe to Zowe ID and recycle the Zowe STC issue got fixed. Now we are able to access the Zowe Editor.

What file owner? I'm getting the same issue every time in Editor, both for USS and MVS files

**Comment by rudatp** [comment]
I just switched the configuration to ATTLS and Editor is working. What could be the reason, that it fails with TLS, but not with ATTLS?
